packages/media/src/types.ts

```diff
@@ -253,7 +253,17 @@ export interface ResourceSize {
   height: number;
 }
 
-export type ResourceId = string;
+// TODO: Enforce number type for dimensions already in wp-* integration packages.
+// TODO: Make sure all properties are required.
+// For this, move logic like normalizeResourceSizes and parts of calculateSrcSet.
```

Comment:
Alternatively, could we use `Partial<ResourceSizeInput>` in [normalizeResourceSizes](https://github.com/GoogleForCreators/web-stories-wp/pull/12167/files#diff-314afaadf6da1291d3cc56473d42741919bfb1971eeefc1a0017d452c6db6d0bR29) typing to keep these required already now? And then once the type is complete, we could cast it to `ResourceSizeInput` when [adding to the normalized sizes](https://github.com/GoogleForCreators/web-stories-wp/pull/12167/files#diff-314afaadf6da1291d3cc56473d42741919bfb1971eeefc1a0017d452c6db6d0bR40).`